CMakeLists.txt

```diff
@@ -69,89 +37,48 @@ set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fno-common -Woverloaded-vir
 
 set (CMAKE_MODULE_LINKER_FLAGS "-Wl,-flat_namespace -Wl,-undefined -Wl,suppress")
 
-macro(add_clang_plugin name)
-   set (srcs ${ARGN})
-
-   include_directories(
-      "${LLVM_INLCUDE_DIR}/"
-      "${CLANG_INCLUDE_DIRS}"
-      "${CMAKE_CURRENT_SOURCE_DIR}/src")
-   link_directories( "${LLVM_LIB_DIR}/" )
-
-   add_library( ${name} SHARED ${srcs} )
-
-   if (SYMBOL_FILE)
-      set_target_properties( ${name} PROPERTIES LINK_FlAGS
-         "-exported_symbols_list ${SYMBOL_FILE}")
-   endif()
+set (USER_LIBS pthread)
 
-   foreach (user_lib ${USER_LIBS})
-      target_link_libraries( ${name} ${user_lib} )
-   endforeach()
+option(INCLUDE_FCCSW_CHECKERS "Turn on to activate FCCSW coding convention checkers.")
+option(INCLUDE_ROOT_CHECKERS "Turn on to activate ROOT coding convention checkers.")
+option(INCLUDE_GENERAL_CHECKERS "Turn on to activate general coding convention checkers.")
```

Comment:
Let's make these a bit shorter, I think jist `FCCSW_CHECKERS` should suffice, since the command line arg looks something like `FCCSW_CHECKERS=ON`... Also, can you put the default `ON` for the general checkers?